session-helper/flatpak-session-helper.c

```diff
@@ -403,6 +404,27 @@ on_name_lost (GDBusConnection *connection,
   exit (1);
 }
 
+typedef struct {
+  const gchar *source;
+  char *real;
+  GFileMonitor *monitor_source;
+  GFileMonitor *monitor_real;
```

Comment:
I think this struct could do with a comment explaining why we need to monitor the target of symlinks, including specific NetworkManager examples. Otherwise this code is all a bit opaque until you’ve read and understood it all.

Reply:
Will do.